python/unblob/file_utils.py

```diff
@@ -355,17 +355,25 @@ def parse(
         return struct_parser(file)
 
 
-def get_endian(file: File, big_endian_magic: int) -> Endian:
-    """Read a four bytes magic and derive endianness from it.
+def get_endian(
+    file: File, big_endian_magic: int, endian_len: Literal[2] | Literal[4] = 4
```

Comment:
Failing the type check for earlier python versions, `Literal[2, 4]` should work, and is even shorter.